### ISSUE TITLE
[feat] Automatically reset security classes when re-interviewing an S2 authenticated device after firmware upgrade

### ISSUE THREAD

**Issue by cmamatey** [zwave-js/zwavejs2mqtt#2601]
**Is your feature request related to a problem? Please describe.**
I just used the OTA firmware upgrade feature for the first time last night with great results; great work contributors!
However when I updated the firmware on an S2 authenticated device, the re-interview process (automatically started when the upgrade finished) did not seem to work correctly. All details for the device were replaced with generic information like "Unknown manufacturer 0xXXXX", and additional re-interviews made no changes.

I learned the fix for me was to perform another interview, but this time to check the box for "Reset security classes". Once re-interviewed with this option selected, the device returned to the expected values and configurations.

**Describe the solution you'd like**
Enhance the firmware upgrade process such that when the device being updated is an S2 authenticated device, the "Reset security classes" option is automatically included during the interview action at the end of the firmware upgrade process. Ideally, this would prevent the device from ever entering the "Unknown manufacturer 0xXXXX" state.

**Describe alternatives you've considered**
None.

**Additional context**
I am unfamiliar with the implementation of S2 and am therefore unware of the potential impacts/risks of resetting security classes by default.

**Comment by jmgiaever** [comment]
I think this can be a cache issue, or that the ID of the device has changed, @AlCalzone ?

This should probably filed in the node-zwavejs repository, but did you have logging (in debud mode) turned on during this process for zwavejs (not zwavejs2mqtt)?

In case you didn't, please turn it on before updating additional devices.

**Comment by AlCalzone** [comment]
Agree, I'm not sure why this should be necessary. This sounds like a corrupted cache file or a bug, but network keys shouldn't change.
